mapie/regression/ccp_regression.py

```diff
@@ -0,0 +1,806 @@
+        ``alpha`` is the complement of the target coverage level.
+
+        By default 0.1
+
+
+    random_state: Optional[int]
+        Integer used to set the numpy seed, to get reproducible calibration
+        results.
+        If ``None``, the prediction intervals will be stochastics, and will
+        change if you refit the calibration (even if no arguments have change).
+
+        WARNING: If ``random_state``is not ``None``, ``np.random.seed`` will
+        be changed, which will reset the seed for all the other random
+        number generators.
+
+        By default ``None``.
+
+    Attributes
+    ----------
+    beta_up: Tuple[NDArray, bool]
+        Calibration fitting results, used to build the upper bound of the
+        prediction intervals.
+        beta_up[0]: Array of shape (phi.n_out, )
+        beta_up[1]: Whether the optimization process converged or not
+                    (the coverage is not garantied if the optimization fail)
+
+    beta_low: Tuple[NDArray, bool]
+        Same as beta_up, but for the lower bound
+
+    References
+    ----------
+    Isaac Gibbs and John J. Cherian and Emmanuel J. Candès.
+    "Conformal Prediction With Conditional Guarantees", 2023
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> from mapie.regression import MapieCCPRegressor
+    >>> X_train = np.array([[0], [1], [2], [3], [4], [5]])
+    >>> y_train = np.array([5, 7.5, 9.5, 10.5, 12.5, 15])
+    >>> mapie_reg = MapieCCPRegressor(alpha=0.1, random_state=1)
+    >>> mapie_reg = mapie_reg.fit_calibrate(
+    ...     X_train,
+    ...     y_train,
+    ... )
+    >>> y_pred, y_pis = mapie_reg.predict(X_train)
+    >>> print(np.round(y_pis[:,:, 0], 2))
+    [[ 4.14  5.57]
+     [ 6.11  7.54]
+     [ 8.07  9.5 ]
+     [10.04 11.46]
+     [12.   13.43]
+     [13.96 15.39]]
+    >>> print(np.round(y_pred, 2))
+    [ 4.86  6.82  8.79 10.75 12.71 14.68]
+    """
+
+    default_sym_ = True
+
+    def __init__(
+        self,
+        estimator: Optional[
+            Union[
+                RegressorMixin,
+                Pipeline,
+                List[Union[RegressorMixin, Pipeline]]
+            ]
+        ] = None,
+        phi: Optional[PhiFunction] = None,
+        cv: Optional[
+            Union[str, BaseCrossValidator, BaseShuffleSplit]
+        ] = "split",
+        alpha: float = 0.1,
+        conformity_score: Optional[ConformityScore] = None,
+        random_state: Optional[int] = None,
+    ) -> None:
+
+        self.random_state = random_state
+        self.cv = cv
+        self.estimator = estimator
+        self.conformity_score_ = conformity_score
+
+        if phi is None:
+            self.phi = PhiFunction(lambda X: np.ones(len(X)))
+        else:
+            self.phi = cast(PhiFunction, phi)
+
+        self.alpha = alpha
+        self.beta_up: Optional[Tuple[NDArray, bool]] = None
+        self.beta_low: Optional[Tuple[NDArray, bool]] = None
+
+    def _check_fit_parameters(self) -> None:
```

Comment:
Good suggestion. Done.

Reply:
Except for the random_state actually, I don't use the sklearn ``check_random_state`` function, because I want to keep the ``int``, and not convert it into a ``RandomState``